kernelci/data/kernelci_api.py

```diff
@@ -60,9 +61,20 @@ def subscribe(self, channel):
         resp = self._post(f'subscribe/{channel}')
         return json.loads(resp.text)['id']
 
+    def subscribe_node_channel(self, filters=None):
+        resp = self._post(f'subscribe/node')
+        sub_id = json.loads(resp.text)['id']
+        self._filters[sub_id] = filters
+        print("Subscribed with filters", self._filters)
+        return sub_id
+
     def unsubscribe(self, sub_id):
         self._post(f'unsubscribe/{sub_id}')
 
+    def unsubscribe_node_channel(self, sub_id):
+        self._filters.pop(sub_id)
+        self._post(f'unsubscribe/{sub_id}')
```

Comment:
I don't think we need a new method for this. It only needs `sub_id`, so we can do something like `self._filters.pop(sub_id) if sub_id in self._filters`

The advantage being that we would have only one method for unsubscribing, so if something needs to always be done to avoid resource leaks we would just have to implement it in that method.

Reply:
Agreed.